packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -120,44 +119,30 @@ export interface RoutedMappings {
 export interface TimelineContentObjectTmp extends TSRTimelineObjBase {
 	inGroup?: string
 }
-/** Max time for initializing devices */
-const INIT_TIMEOUT = 10000
-
-enum DeviceAction {
-	ADD = 'add',
-	READD = 'readd',
-	REMOVE = 'remove',
-}
-
-type DeviceActionResult = {
-	success: boolean
-	deviceId: string
-	action: DeviceAction
+/** Max time for device-related jobs */
+const JOB_TIMEOUT = 10000
+
+enum JobFailure {
+	ADD_ERROR = 'add_error',
+	ADD_TIMEOUT = 'add_timeout',
+	REMOVE_ERROR = 'remove_error',
+	REMOVE_TIMEOUT = 'remove_timeout',
 }
 
-type UpdateDeviceOperationsResult =
-	| {
-			success: true
-			results: DeviceActionResult[]
-	  }
-	| {
-			success: false
-			reason: 'timeout' | 'error'
-			details: string[]
-	  }
-
 /**
  * Represents a connection between Gateway and TSR
  */
 export class TSRHandler {
 	logger: Logger
-	tsr!: Conductor
+	private _tsr!: Conductor
 	// private _config: TSRConfig
 	private _coreHandler!: CoreHandler
 	private _triggerupdateExpectedPlayoutItemsTimeout: any = null
 	private _coreTsrHandlers: { [deviceId: string]: CoreTSRDeviceHandler } = {}
 	private _observers: Array<any> = []
 	private _cachedStudioId = ''
+	private _jobQueueManager: JobQueueManager
+	private _deviceJobFailureStatuses = new Map<string, JobFailure>()
 
 	private _initialized = false
 	private _multiThreaded: boolean | null = null
```

Comment:
What benefit does these fields have being of type `boolean | null` over say `boolean | undefined`?
Is a null check not more processing costly than a undefined check?

Also aren't these boolean values missing the `isX`, or `doesX` naming scheme to better indicate that they are booleans?

Reply:
> What benefit does these fields have being of type boolean | null over say boolean | undefined?

Semantically `boolean | undefined` would make more sense.

> Is a null check not more processing costly than a undefined check?

If anything, it might be faster https://www.measurethat.net/Benchmarks/Show/6389/0/compare-comparison-with-null-or-undefined